### ISSUE TITLE
(v2.0.0) [Semantical Error] The class "Symfony\Contracts\Service\Attribute\Required" is not annotated with @Annotation.

### ISSUE THREAD

**Issue by alexislefebvre** [doctrine/annotations#474]
For your information, `doctrine/annotations` 2 trigger an error with some Symfony installations.

Here is a reproducer: https://github.com/symfony/symfony/issues/48792#issuecomment-1370306674

**Comment by derrabus** [comment]
And this happens with 2.0 only, not with 1.14?

**Comment by greg0ire** [comment]
<details>
<summary>the stack trace of the error</summary>

```

$ bin/console ca:cl -vvv

 // Clearing the cache for the dev environment with debug true

 // Warming up cache...


In FileLoader.php line 178:

  [Symfony\Component\Config\Exception\LoaderLoadException]
  [Semantical Error] The class "Symfony\Contracts\Service\Attribute\Required" is not annotated with @Annotation.
  Are you sure this class can be used as annotation?
  If so, then you need to add @Annotation to the _class_ doc comment of "Symfony\Contracts\Service\Attribute\Required".
  If it is indeed no annotation, then you need to add @IgnoreAnnotation("required") to the _class_ doc comment of method Symfony\Bundle\Fr
  ameworkBundle\Controller\AbstractController::setContainer() in {"path":"..\/src\/Controller\/","namespace":"App\\Controller"} (which is
  being imported from "/tmp/symfony_bug_app_48792/config/routes.yaml"). Make sure there is a loader supporting the "attribute" type.


Exception trace:
  at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/FileLoader.php:178
 Symfony\Component\Config\Loader\FileLoader->doImport() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/FileLoader.php:94
 Symfony\Component\Config\Loader\FileLoader->import() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/YamlFileLoader.php:208
 Symfony\Component\Routing\Loader\YamlFileLoader->parseImport() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/YamlFileLoader.php:99
 Symfony\Component\Routing\Loader\YamlFileLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/FileLoader.php:163
 Symfony\Component\Config\Loader\FileLoader->doImport() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/FileLoader.php:94
 Symfony\Component\Config\Loader\FileLoader->import() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/Configurator/RoutingConfigurator.php:45
 Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator->import() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Kernel/MicroKernelTrait.php:81
 App\Kernel->configureRoutes() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Kernel/MicroKernelTrait.php:202
 App\Kernel->loadRoutes() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/ObjectLoader.php:55
 Symfony\Component\Routing\Loader\ObjectLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/DelegatingLoader.php:37
 Symfony\Component\Config\Loader\DelegatingLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Routing/DelegatingLoader.php:67
 Symfony\Bundle\FrameworkBundle\Routing\DelegatingLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Routing/Router.php:65
 Symfony\Bundle\FrameworkBundle\Routing\Router->getRouteCollection() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Router.php:317
 Symfony\Component\Routing\Router->getMatcherDumperInstance() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Router.php:255
 Symfony\Component\Routing\Router->Symfony\Component\Routing\{closure}() at /tmp/symfony_bug_app_48792/vendor/symfony/config/ResourceCheckerConfigCacheFactory.php:36
 Symfony\Component\Config\ResourceCheckerConfigCacheFactory->cache() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Router.php:263
 Symfony\Component\Routing\Router->getMatcher() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Routing/Router.php:92
 Symfony\Bundle\FrameworkBundle\Routing\Router->warmUp() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/CacheWarmer/RouterCacheWarmer.php:42
 Symfony\Bundle\FrameworkBundle\CacheWarmer\RouterCacheWarmer->warmUp() at /tmp/symfony_bug_app_48792/vendor/symfony/http-kernel/CacheWarmer/CacheWarmerAggregate.php:96
 Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerAggregate->warmUp() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:236
 Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->warmup() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:148
 Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->execute() at /tmp/symfony_bug_app_48792/vendor/symfony/console/Command/Command.php:312
 Symfony\Component\Console\Command\Command->run() at /tmp/symfony_bug_app_48792/vendor/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Console/Application.php:88
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /tmp/symfony_bug_app_48792/vendor/symfony/console/Application.php:314
 Symfony\Component\Console\Application->doRun() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Console/Application.php:77
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /tmp/symfony_bug_app_48792/vendor/symfony/console/Application.php:168
 Symfony\Component\Console\Application->run() at /tmp/symfony_bug_app_48792/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:54
 Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at /tmp/symfony_bug_app_48792/vendor/autoload_runtime.php:29
 require_once() at /tmp/symfony_bug_app_48792/bin/console:11

In AnnotationException.php line 36:

  [Doctrine\Common\Annotations\AnnotationException]
  [Semantical Error] The class "Symfony\Contracts\Service\Attribute\Required" is not annotated with @Annotation.
  Are you sure this class can be used as annotation?
  If so, then you need to add @Annotation to the _class_ doc comment of "Symfony\Contracts\Service\Attribute\Required".
  If it is indeed no annotation, then you need to add @IgnoreAnnotation("required") to the _class_ doc comment of method Symfony\Bundle\Fr
  ameworkBundle\Controller\AbstractController::setContainer().


Exception trace:
  at /tmp/symfony_bug_app_48792/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationException.php:36
 Doctrine\Common\Annotations\AnnotationException::semanticalError() at /tmp/symfony_bug_app_48792/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/DocParser.php:833
 Doctrine\Common\Annotations\DocParser->Annotation() at /tmp/symfony_bug_app_48792/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/DocParser.php:708
 Doctrine\Common\Annotations\DocParser->Annotations() at /tmp/symfony_bug_app_48792/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/DocParser.php:368
 Doctrine\Common\Annotations\DocParser->parse() at /tmp/symfony_bug_app_48792/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationReader.php:206
 Doctrine\Common\Annotations\AnnotationReader->getMethodAnnotations() at /tmp/symfony_bug_app_48792/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/PsrCachedReader.php:155
 Doctrine\Common\Annotations\PsrCachedReader->fetchFromCache() at /tmp/symfony_bug_app_48792/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/PsrCachedReader.php:119
 Doctrine\Common\Annotations\PsrCachedReader->getMethodAnnotations() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/AnnotationClassLoader.php:364
 Symfony\Component\Routing\Loader\AnnotationClassLoader->getAnnotations() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/AnnotationClassLoader.php:129
 Symfony\Component\Routing\Loader\AnnotationClassLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/Loader.php:48
 Symfony\Component\Config\Loader\Loader->import() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/Psr4DirectoryLoader.php:90
 Symfony\Component\Routing\Loader\Psr4DirectoryLoader->loadFromDirectory() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/Psr4DirectoryLoader.php:46
 Symfony\Component\Routing\Loader\Psr4DirectoryLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/FileLoader.php:142
 Symfony\Component\Config\Loader\FileLoader->doImport() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/FileLoader.php:94
 Symfony\Component\Config\Loader\FileLoader->import() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/YamlFileLoader.php:208
 Symfony\Component\Routing\Loader\YamlFileLoader->parseImport() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/YamlFileLoader.php:99
 Symfony\Component\Routing\Loader\YamlFileLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/FileLoader.php:163
 Symfony\Component\Config\Loader\FileLoader->doImport() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/FileLoader.php:94
 Symfony\Component\Config\Loader\FileLoader->import() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/Configurator/RoutingConfigurator.php:45
 Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator->import() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Kernel/MicroKernelTrait.php:81
 App\Kernel->configureRoutes() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Kernel/MicroKernelTrait.php:202
 App\Kernel->loadRoutes() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Loader/ObjectLoader.php:55
 Symfony\Component\Routing\Loader\ObjectLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/config/Loader/DelegatingLoader.php:37
 Symfony\Component\Config\Loader\DelegatingLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Routing/DelegatingLoader.php:67
 Symfony\Bundle\FrameworkBundle\Routing\DelegatingLoader->load() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Routing/Router.php:65
 Symfony\Bundle\FrameworkBundle\Routing\Router->getRouteCollection() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Router.php:317
 Symfony\Component\Routing\Router->getMatcherDumperInstance() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Router.php:255
 Symfony\Component\Routing\Router->Symfony\Component\Routing\{closure}() at /tmp/symfony_bug_app_48792/vendor/symfony/config/ResourceCheckerConfigCacheFactory.php:36
 Symfony\Component\Config\ResourceCheckerConfigCacheFactory->cache() at /tmp/symfony_bug_app_48792/vendor/symfony/routing/Router.php:263
 Symfony\Component\Routing\Router->getMatcher() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Routing/Router.php:92
 Symfony\Bundle\FrameworkBundle\Routing\Router->warmUp() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/CacheWarmer/RouterCacheWarmer.php:42
 Symfony\Bundle\FrameworkBundle\CacheWarmer\RouterCacheWarmer->warmUp() at /tmp/symfony_bug_app_48792/vendor/symfony/http-kernel/CacheWarmer/CacheWarmerAggregate.php:96
 Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerAggregate->warmUp() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:236
 Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->warmup() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:148
 Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->execute() at /tmp/symfony_bug_app_48792/vendor/symfony/console/Command/Command.php:312
 Symfony\Component\Console\Command\Command->run() at /tmp/symfony_bug_app_48792/vendor/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Console/Application.php:88
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /tmp/symfony_bug_app_48792/vendor/symfony/console/Application.php:314
 Symfony\Component\Console\Application->doRun() at /tmp/symfony_bug_app_48792/vendor/symfony/framework-bundle/Console/Application.php:77
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /tmp/symfony_bug_app_48792/vendor/symfony/console/Application.php:168
 Symfony\Component\Console\Application->run() at /tmp/symfony_bug_app_48792/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:54
 Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at /tmp/symfony_bug_app_48792/vendor/autoload_runtime.php:29
 require_once() at /tmp/symfony_bug_app_48792/bin/console:11

cache:clear [--no-warmup] [--no-optional-warmers]
```

</details>

This appears to be related to routing, and to this annotation: https://github.com/symfony/symfony/blob/e4117121a7bd1f7e07961b12e25b72bd0c03db6c/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php#L62 . Why is it lowercase? The class indeed misses the `@Annotation` annotation, so I'm not sure why  this works with `doctrine/annotations` v1.

**Comment by derrabus** [comment]
`@required` is _not_ a Doctrine annotation and it's not meant to reference the `Required` attribute class. The expected result would be that Doctrine Annotations does not find any annotations worth evaluating and does nothing.

**Comment by greg0ire** [comment]
So it's an annotation, but not a Doctrine one, and not merely a phpdoc annotation I suppose (right?).

Is what the error messages suggest worth considering?

> If it is indeed no annotation, then you need to add @IgnoreAnnotation("required")

**Comment by derrabus** [comment]
That's something that Symfony would need to apply to `AbstractController`, right?

**Comment by greg0ire** [comment]
Yes (I just tested with the reproducer, it does fix the issue).

**Comment by alexislefebvre** [comment]
> This appears to be related to routing, and to this annotation: https://github.com/symfony/symfony/blob/e4117121a7bd1f7e07961b12e25b72bd0c03db6c/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php#L62 .

I confirm that the issue is caused by the annotation + attribute:

```
    /**
     * @required
     */
    #[Required]
```

Source: https://github.com/symfony/symfony/blob/e4117121a7bd1f7e07961b12e25b72bd0c03db6c/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php#L61-L64

Is it worth adding a test of this config in this repo? Is it an edge case that annotations should support?

--- 

Update: my bad, removing ` #[Required]` doesn't fix the issue, `@required` is the root of the issue here, not the fact that there is an annotation and an attribute.

**Comment by alexislefebvre** [comment]
Fixed by https://github.com/symfony/symfony/pull/48916

**Comment by j-mahlke** [comment]
This issue is still open for the current symfony 6.2 and needs a merge, as 6.3 release is still some time in the future. Can somebody confirm this?

**Comment by alexislefebvre** [comment]
@jmahlkeXTAIN the issue will be fixed in the next releases of Symfony 5.4 and Symfony 6.0 to 6.2. Meanwhile, use `doctrine/annotations` v1.
